R/GeneSetAnalysis.R

```diff
@@ -0,0 +1,77 @@
+#' @import ggplot2
+
+
+#' @title PathwayEnrichment
+#'
+#' @description Runs GSEA analysis on a seurat object
+#' @param seuratObj The Seurat object containing the data. 
+#' @param groupField Field on which to group to calculate AUC
+#' @param msigdbSpecies MSigDBSpecies name to retrieve gene sets from. Run msigdbr_species() to see available species.
+#' @param msigdbCategory MSigDB collection abbreviation, such as H or C1. Run msigdbr_collections() to see available collections.
+#' @param scoreType This parameter defines the GSEA score type.Possible options are("std","pos", "neg")
+#' @param selectedPathways A set of pathways to test.
+#' @param msigdbSubcategory MSigDB sub-collection abbreviation, such as CGP or BP
+#' @return A seurat object with NES values for each pathway appended to each cell 
+#' @export
+
+PathwayEnrichment <- function(seuratObj,
+                              groupField,
+                              msigdbSpecies = "Macaca mulatta",
```

Comment:
@maanasa-kaza to test human vs macaque annotations, we could test (offline) some of the stims for a TCR activation pathway. It's probably close to the sweet spot of ground truth and what we would care about running headlessly on prime-seq.